### PR TITLE
boards/opencm904: remap uart pins

### PR DESCRIPTION
**Pull request by astralien3000**
The opencm904 need the USART1 to be mapped on alternativ pins to use the dynamixel TTL bus.
The periph_uart driver is not able to "remap" uart pins when needed.
So that, I need to do that in `board_init`.

### CODE REVIEW

**Comment by astralien3000** [comment]
I also added a DXL_DIR_PIN definition, which is used for the Dynamixel TTL bus

**Comment by astralien3000** [comment]
Finally ! Can someone review this, please ?

**Comment by aabadie** [comment]
Changes looks good. How can this be tested ?

**Comment by astralien3000** [comment]
Hmm, not simple.
I tested it with PR #6725 and dynamixel servomotors, but it is an other PR.

But I think you can test it with the 3 pins named D,X,L on the board : 
 - D : Relevant to Dynamixel TTL Bus and used to select TX, RX (direction pin)
 - X : Dynamixel TTL Bus’s TX pin
 - L : Dynamixel TTL Bus’s RX pin

You can at least test uart with tests/periph_uart and the 2 last pins, I think.
To test the direction pin, you will have to do more "weird" things.

File: boards/opencm904/board.c [hunk 1]
```diff
@@ -22,6 +22,18 @@
 #include "periph/gpio.h"
 #include "log.h"
 
+#define AFIO_REMAP_USE_MAPR2 (1U << 31)
+#define AFIO_REMAP_USART1 AFIO_MAPR_USART1_REMAP
+
+static inline void _afio_remap(uint32_t remapping) {
+  if (remapping & AFIO_REMAP_USE_MAPR2) {
```
Comment (Kijewski) [inline boards/opencm904/board.c]: I would simply remove the `if` test. Always nand remapping with the flag.

**Comment by astralien3000** [comment]
>```
>static inline void _afio_remap(uint32_t remapping) {
>  if (remapping & AFIO_REMAP_USE_MAPR2) {
>```
>I would simply remove the if test. Always nand remapping with the flag.

I got this by email, but I can't find it on github...
Anyway, I simplified the remaping, as asked. (It's better than using copy/pasted "dirty" code)

File: boards/opencm904/include/board.h [hunk 2]
```diff
@@ -55,11 +55,20 @@ extern "C" {
 /** @} */
 
 /**
- * @brief   User button
+ * @brief User button
  */
 #define BTN_B1_PIN          GPIO_PIN(PORT_C, 15)
 
 /**
+ * @brief Pin used to switch RX and TX mode for the Dynamixel TTL bus
```
Comment (aabadie) [inline boards/opencm904/include/board.h]: This block of code doesn't seem to be related to this change but rather to #6725. Can you revert it and backport to the related PR ?

Comment (aabadie) [inline boards/opencm904/include/board.h]: my bad, it seems that this is for the Dynamixel TTL bus which is available on that board.

**Comment by aabadie** [comment]
all green, go!
